.github/workflows/cpp_package.yaml

```diff
@@ -60,5 +68,5 @@ jobs:
     - name: "Upload C++ packages"
       uses: actions/upload-artifact@v4
       with:
-          name: cpp_bin
+          name: cpp_bin-${{ matrix.package_suffix }}
```

Comment:
The cmake step from `ci.yaml` uses `cpp_bin-${{ matrix.os }}`
It might make sense to use the same, although i agree it's more error prone to changes in a location that is not tested until the release comes.

Reply:
I thought about matrix.os, but I'm worried that when we change the base os (from say ubuntu-20.04 to ubuntu-23.04), we'll forget to adjust the release step.